dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/service/TriMetadataServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.rpc.protocol.tri.service;
+
+import org.apache.dubbo.common.utils.CollectionUtils;
+import org.apache.dubbo.metadata.MetadataInfo;
+import org.apache.dubbo.metadata.MetadataService;
+import org.apache.dubbo.rpc.model.ApplicationModel;
+import org.apache.dubbo.triple.metadata.AllMetaRequest;
+import org.apache.dubbo.triple.metadata.AllMetaResponse;
+import org.apache.dubbo.triple.metadata.DubboMetadataTriple;
+import org.apache.dubbo.triple.metadata.MetaRequest;
+import org.apache.dubbo.triple.metadata.MetaResponse;
+import org.apache.dubbo.triple.metadata.ResponseStatus;
+import org.apache.dubbo.triple.metadata.ServiceInfo;
+
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This implemented class provides a point-to-point built-in metadata
+ * retrieval service by encapsulating Dubbo's MetadataService.
+ */
+public class TriMetadataServiceImpl extends DubboMetadataTriple.MetadataImplBase {
+
+    /**
+     * Retrieves metadata information for a specified revision.
+     *
+     * @param metaRequest the request for metadata information
```

Comment:
Move this built-in service to `dubbo-metrics-api`, and change exporter to use proto.